cpp/src/io/parquet/page_data.cu

```diff
@@ -539,7 +485,9 @@ inline __device__ void gpuOutputString(volatile page_state_s* s, int src_pos, vo
   }
   if (s->dtype_len == 4) {
     // Output hash
-    *static_cast<uint32_t*>(dstv) = device_str2hash32(ptr, len);
+    uint32_t constexpr hash_seed = 33;
```

Comment:
Why do we use `33` here? Does it have any special meaning?

Reply:
I suspect that somebody else may have the same question, if not now then later when reading the code. So regardless if this is just some random number or not, please add a comment line saying so.

Reply:
That’s a great question. I don’t know the answer but I will try to find out more via git blame and perhaps reading the Parquet source.

Reply:
I could only find the original PR https://github.com/rapidsai/cudf/pull/1235 introduced this value, but I don't have a reference in the upstream Parquet code base for how this is used. The Murmur3 hash is defined in [parquet-cpp/src/parquet/murmur3.h](https://github.com/apache/parquet-cpp/blob/642da055adf009652689b20e68a198cffb857651/src/parquet/murmur3.h) but I don't see this seed being used.

Reply:
So the seed there is `DEFAULT_SEED = 1361930890`. Maybe it is just a random value...

Reply:
Still recommend adding a comment line for the seed value anyway.

Reply:
@ttnghia @vuule It looks like this path is only taken if `strings_to_categoricals` is enabled, which results in the string being discarded in favor of its hash value. (It is somewhat concerning that it took great effort to identify that this is decided by an option controlled at a much higher level...) I added a comment to clarify.